qemu/tests/rh_kselftests_vm.py

```diff
@@ -0,0 +1,48 @@
+from avocado.core import exceptions
+
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    rh_kselftests_vm test
+    1) Download the current kernel selftests RPM
+    2) Install the RPM
+    3) Execute the kernel selftests
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login()
+    kernel_path = params.get("kernel_path", "/tmp/kernel")
+    tests_execution_cmd = params.get("tests_execution_cmd")
+
+    session.cmd("mkdir -p %s" % kernel_path)
+    kernel_version = session.cmd_output("uname -r").strip().split("+")[0]
+    error_context.base_context("The kernel version: %s" % kernel_version, test.log.info)
+
+    error_context.context("Download the kernel selftests RPM", test.log.debug)
+    session.cmd("cd %s" % kernel_path)
+    session.cmd(
+        "brew download-build --rpm kernel-selftests-internal-%s.rpm" % kernel_version,
+        180,
+    )
+
+    error_context.context("Install the RPM", test.log.debug)
+    session.cmd("dnf install -y ./kernel-*")
+
+    error_context.base_context("Execute the selftests", test.log.info)
+    s, o = session.cmd_status_output(tests_execution_cmd, 180)
+
+    # Exit code for skipped selftests is 4, raise a warning until is fixed
+    if s == 4:
```

Comment:
I do not think it is a issue need be fixed in kernel selftests, but it is intentional.

From my checking with the source code,  if there is a skip case after a fail case, the return value is 4.  It is wrong(Not the scope of this PR but the kselftest).

Suggest to use whitelist for the know test can be skipped, and parse the output of the test to make a final decision of the test case result.

Reply:
@yfu but then how will the user know that there are skipped tests? If we set the skipped as passed it's possible the user won't be aware, right?

Reply:
We are talking about exitcode in this thread.
```bash
run_test() {
	if test_selected ${CATEGORY}; then
		# On memory constrainted systems some tests can fail to allocate hugepages.
		# perform some cleanup before the test for a higher success rate.
		if [ ${CATEGORY} == "thp" ] | [ ${CATEGORY} == "hugetlb" ]; then
			echo 3 > /proc/sys/vm/drop_caches
			sleep 2
			echo 1 > /proc/sys/vm/compact_memory
			sleep 2
		fi

		local test=$(pretty_name "$*")
		local title="running $*"
		local sep=$(echo -n "$title" | tr "[:graph:][:space:]" -)
		printf "%s\n%s\n%s\n" "$sep" "$title" "$sep" | tap_prefix

		("$@" 2>&1) | tap_prefix
		local ret=${PIPESTATUS[0]}
		count_total=$(( count_total + 1 ))
		if [ $ret -eq 0 ]; then
			count_pass=$(( count_pass + 1 ))
			echo "[PASS]" | tap_prefix
			echo "ok ${count_total} ${test}" | tap_output
		elif [ $ret -eq $ksft_skip ]; then
			count_skip=$(( count_skip + 1 ))
			echo "[SKIP]" | tap_prefix
			echo "ok ${count_total} ${test} # SKIP" | tap_output
			exitcode=$ksft_skip
		else
			count_fail=$(( count_fail + 1 ))
			echo "[FAIL]" | tap_prefix
			echo "not ok ${count_total} ${test} # exit=$ret" | tap_output
			exitcode=1
		fi
	fi # test_selected
}
```

In the loop, 
|         | 1: PASS | 1: SKIP | 1: FAIL |
|:-------:|:-------:|:-------:|:-------:|
| 2: PASS |    0    |    4    |    1    |
| 2: SKIP |    4    |    4    |    4    |
| 2: FAIL |    1    |    1    |    1    | 

Scenarios that we need to agree on are `SKIP --> FAIL(exit 1)` and `FAIL --> SKIP(exit 4)`. I think `FAIL --> SKIP(exit 4)` must return 1 but this needs to be changed at Linux source code, for now, I don't think we have good solution to handle it. A workaround is to check the output and collect count of `[PASS]` `[SKIP]` `[FAIL]` rather than check the exit code. Keep `if s == 4:` is this version is in order to safely raise a warning if we will have new test cases for mm in the future.

Reply:
> @yfu but then how will the user know that there are skipped tests? If we set the skipped as passed it's possible the user won't be aware, right?

Any skip but case not in whitelist should fail this auto case.  Checking the output is needed.
1. Use exitcode is incorrect if there are skip case but after a fail one
2. If the skip case is in whitelist,  i prefer to mark test case status to PASS instead of WARN as it is a known issue.

Reply:
Added the whitelist approach